Sources/TuistCache/Cache/URL+Cache.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+import TuistCore
+
+enum CacheURLError: LocalizedError {
+    case missingCloudConfig
+    case incorrectCloudConfig
+}
+
+extension URL {
+    static func apiCacheURL(hash: String, config: Config, contentMD5: String? = nil) throws -> URL {
+        guard let cloudConfig = config.cloud else { throw CacheURLError.missingCloudConfig }
```

Comment:
I find it odd that we are bubbling up the error from a method so deep in the stack. I'd not expect a method like this one to validate whether cloud is configured or not. In fact, if the cloud is not configured, I don't think we should treat it as an error in the first place. The user simply doesn't want to use the cloud.

That being said, I'd change the signature to be:

```swift
static func apiCacheURL(hash: String, cloudURL: URL, contentMD5: String) throws -> URL
```